mempool/mempool.go

```diff
@@ -91,6 +91,15 @@ type Mempool interface {
 	// trigger once every height when transactions are available.
 	EnableTxsAvailable()
 
+	// GetTxByKey gets a tx by its key from the mempool cache. Returns the tx and a bool indicating its presence in the tx cache.
```

Comment:
[question] should this clarify that it's referring to the mempool tx cache? Or get rid of cache entirely? "cache" seems ambiguous because there is now a tx cache and an evicted tx cache.

```suggestion
	// GetTxByKey gets a tx by its key from the mempool tx cache. Returns the tx and a bool indicating its presence in the tx cache.
```

```suggestion
	// GetTxByKey gets a tx by its key from the mempool. Returns the tx and a bool indicating its presence in the tx cache.
```